File: app/src/main/java/com/androidexamples/app/database/ExampleDB.java
package com.androidexamples.app.database;

import android.content.ContentValues;
import android.content.Context;
import android.database.sqlite.SQLiteDatabase;

/**
 * Created by marcosvbras on 22/01/17.
 */

public class ExampleDB {

    private int id;
    private String name;
    private double weight;
    private boolean active;
    private TypeDB type;

    public ExampleDB() {}

    public ExampleDB(int id, String name, double weight, boolean active, TypeDB type) {
        this.id = id;
        this.name = name;
        this.weight = weight;
        this.active = active;
        this.type = type;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public double getWeight() {
        return weight;
    }

    public void setWeight(double weight) {
        this.weight = weight;
    }

    public boolean isActive() {
        return active;
    }

    public void setActive(boolean active) {
        this.active = active;
    }

    /**
     * Database section */

    public static final String TABLE_NAME = "Example";
    public static final String COLUMN_ID = "_id";
    public static final String COLUMN_NAME = "name";
    public static final String COLUMN_WEIGHT = "weight";
    public static final String COLUMN_ACTIVE = "active";
    public static final String COLUMN_TYPE_ID = "type_id";

    public static final String TABLE_CREATE = "CREATE TABLE IF NOT EXISTS " + TABLE_NAME + " ("
            + COLUMN_ID + " INTEGER PRIMARY KEY, "
            + COLUMN_NAME + " TEXT, "
            + COLUMN_WEIGHT + " DOUBLE, "
            + COLUMN_ACTIVE + " INTEGER, "
            + COLUMN_TYPE_ID + " INTEGER, "
            + "FOREIGN KEY (" + COLUMN_TYPE_ID + ") REFERENCES " + TypeDB.TABLE_NAME + "(" + COLUMN_ID + ")"
            + ");";

    public boolean save(Context context) {
        ExampleDatabase database = new ExampleDatabase(context);
        ContentValues contentValues = new ContentValues();
        contentValues.put(COLUMN_NAME, name);
        contentValues.put(COLUMN_WEIGHT, weight);
        contentValues.put(COLUMN_ACTIVE, active);
        contentValues.put(COLUMN_TYPE_ID, type.getId());
        long result = database.getDatabase().insert(TABLE_NAME, null, contentValues);
        database.closeConnection();
        return result == 1;
    }

    public void getLastEntry() {

    }
}

File: app/src/main/java/com/androidexamples/app/database/ExampleDB.kt
package com.androidexamples.app.database

import android.content.ContentValues
import android.content.Context
import android.database.sqlite.SQLiteDatabase

/**
 * Created by marcosvbras on 22/01/17.
 */

class ExampleDB {

    var id: Int = 0
    var name: String? = null
    var weight: Double = 0.toDouble()
    var isActive: Boolean = false
    private var type: TypeDB? = null

    constructor() {}

    constructor(id: Int, name: String, weight: Double, active: Boolean, type: TypeDB) {
        this.id = id
        this.name = name
        this.weight = weight
        this.isActive = active
        this.type = type
    }

    fun save(context: Context): Boolean {
        val database = ExampleDatabase(context)
        val contentValues = ContentValues()
        contentValues.put(COLUMN_NAME, name)
        contentValues.put(COLUMN_WEIGHT, weight)
        contentValues.put(COLUMN_ACTIVE, isActive)
        contentValues.put(COLUMN_TYPE_ID, type?.id)
        val result = database.database.insert(TABLE_NAME, null, contentValues)
        database.closeConnection()
        return result == 1L
    }

    fun getLastEntry() {

    }

    companion object {

        /**
         * Database section  */

        val TABLE_NAME = "Example"
        val COLUMN_ID = "_id"
        val COLUMN_NAME = "name"
        val COLUMN_WEIGHT = "weight"
        val COLUMN_ACTIVE = "active"
        val COLUMN_TYPE_ID = "type_id"

        val TABLE_CREATE = ("CREATE TABLE IF NOT EXISTS " + TABLE_NAME + " ("
                + COLUMN_ID + " INTEGER PRIMARY KEY, "
                + COLUMN_NAME + " TEXT, "
                + COLUMN_WEIGHT + " DOUBLE, "
                + COLUMN_ACTIVE + " INTEGER, "
                + COLUMN_TYPE_ID + " INTEGER, "
                + "FOREIGN KEY (" + COLUMN_TYPE_ID + ") REFERENCES " + TypeDB.TABLE_NAME + "(" + COLUMN_ID + ")"
                + ");")
    }
}
